Tella/Data/Networking/Repositories/UwaziServerRepositories.swift

```diff
@@ -56,19 +57,64 @@ struct UwaziServerRepository: WebRepository {
     }
 
     func checkServerURL(serverURL: String) -> AnyPublisher<UwaziCheckURL, APIError> {
-        let apiResponse :  APIResponse<UwaziCheckURLDTO> = getAPIResponse(endpoint: API.checkURL(serverURL: serverURL))
+        let apiResponse: APIResponse<UwaziCheckURLDTO> = getAPIResponse(endpoint: API.checkURL(serverURL: serverURL))
         return apiResponse
             .compactMap{$0.0.toDomain() as? UwaziCheckURL}
             .eraseToAnyPublisher()
     }
 
     func getLanguage(serverURL: String) -> AnyPublisher<UwaziLanguage, APIError> {
-        let apiResponse :  APIResponse<UwaziLanguageDTO> = getAPIResponse(endpoint: API.getLanguage(serverURL: serverURL))
+        let apiResponse: APIResponse<UwaziLanguageDTO> = getAPIResponse(endpoint: API.getLanguage(serverURL: serverURL))
         return apiResponse
             .compactMap{$0.0.toDomain() as? UwaziLanguage}
             .eraseToAnyPublisher()
     }
 
+    ///  Get all the templetes related to the Uwazi server
+    /// - Parameters:
+    ///   - serverURL: the URL of the server
+    ///   - cookieList:  The array of string which consist of  token and the locale of the selected uwazi server if public instance then the array is empty
+    /// - Returns: AnyPublisher with UwaziTemplateResult or APIError if any
+    func getTemplate(serverURL: String, cookieList: [String]) -> AnyPublisher<UwaziTemplate, APIError> {
+        let apiResponse: APIResponse<UwaziTemplateDTO> = getAPIResponse(endpoint: API.getTemplate(serverURL: serverURL, cookieList: cookieList))
+        return apiResponse
+            .compactMap{$0.0.toDomain() as? UwaziTemplate}
```

Comment:
I think in this level we don't need to convert the DTO to domain because we will use it later to handle the allowedPublicTemplates and translation ...